specs/addIndex.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { describe, it, expect } from "./_describe.ts"
+import { addIndex, map, add, multiply, reduce, sum } from '../mod.ts'
+import { eq } from "./utils/utils.ts"
+
+describe('addIndex', () => {
+  
+  const list = [4,
+    'f',
+    undefined,
+    NaN,
+    Infinity,
+    10
+  ]
+
+  const indexedMap = addIndex(map)
+
+  const indexedReduce = addIndex(reduce)
+
+  let sumArr = (tot: number, num: number) => { return tot + num}
+
+  let squareEnds = (x: any, idx: number, list: ArrayLike<any>) => {
+    return (idx === 0 || idx === list.length - 1) ? x * x : x;
+  };
+
+  it('should work as normal map function', () => {   
+    eq(indexedMap(multiply(2))(list), [8, NaN, NaN, NaN, Infinity, 20])
+  })
+  it('should pass second param as index', () => {
+    eq(indexedMap(add)(list), [4, 'f1', NaN, NaN, Infinity, 15])
+  })
+  it('should pass params in order: iteratorFunc, index, list', () => {
+    let makeSquareEnds = indexedMap(squareEnds);
```

Comment:
the test is incomplete...because there is no number in the array but at the ends

Reply:
Changing array to have numbers in middle as well.